polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -161,14 +163,15 @@ void MLIRScanner::initSupportedFunctions() {
                         "5rangeEE3getILi0EEENS3_ILi1EEEv");
 }
 
-void MLIRScanner::init(mlir::func::FuncOp function, const FunctionDecl *fd) {
-  this->function = function;
-  this->EmittingFunctionDecl = fd;
+void MLIRScanner::init(mlir::func::FuncOp func, const FunctionDecl *fd) {
+  assert(fd && "Expecting valid function declaration");
 
-  if (ShowAST) {
-    llvm::errs() << "Emitting fn: " << function.getName() << "\n";
-    llvm::errs() << *fd << "\n";
-  }
+  function = func;
+  EmittingFunctionDecl = fd;
+
+  if (ShowAST)
+    llvm::dbgs() << "Emitting fn: " << function.getName() << "\n"
+                 << *fd << "\n";
```

Comment:
`llvm::errs` is used for this in the rest of the file, maybe we should stick to it.